File: src-list.pl
#!/usr/bin/perl
use XML::XPath;
use XML::XPath::XMLParser;

my $xp = XML::XPath->new(filename => 'EWARM/mksRobinNano.ewp');
    
if (($ARGV[0] eq "src") or (not $ARGV[0]))  {
	my $nodeset = $xp->find('//file[not(excluded[configuration="mksRobinNano"])]/name/text()'); 
    foreach my $node ($nodeset->get_nodelist) {
		my $f=XML::XPath::XMLParser::as_string($node);
		$f =~ s/\$PROJ_DIR\$\\..\\//;
		$f =~ s/\\/\//g;
		if (-e $f) {
			print "$f\n"; 
		}
	}
} elsif ($ARGV[0] eq "define")  {
	my $nodeset = $xp->find('//settings/data/option[name="CCDefines"]/state/text()');
    foreach my $node ($nodeset->get_nodelist) {
		my $f=XML::XPath::XMLParser::as_string($node);
        print "$f\n"; 
	}	
} elsif ($ARGV[0] eq "include")  {
	my $nodeset = $xp->find('//settings/data/option[name="CCIncludePath2"]/state/text()');
    foreach my $node ($nodeset->get_nodelist) {
		my $f=XML::XPath::XMLParser::as_string($node);
		$f =~ s/\$PROJ_DIR\$\\..\\//;
		$f =~ s/\$PROJ_DIR\$\/..\///;
		$f =~ s/\\/\//g;
        print "$f\n"; 
	}	
} elsif ($ARGV[0] eq "lib")  {
	my $nodeset = $xp->find('//settings/data/option[name="IlinkAdditionalLibs"]/state/text()');
    foreach my $node ($nodeset->get_nodelist) {
		my $f=XML::XPath::XMLParser::as_string($node);
		$f =~ s/\$PROJ_DIR\$\\..\\//;
		$f =~ s/\$PROJ_DIR\$\/..\///;
		$f =~ s/\\/\//g;
        print "$f\n"; 
	}	
}
	

File: src-list.py
import libxml2
import os.path
from sys import argv

doc = libxml2.parseFile('EWARM/mksRobinNano.ewp')
ctxt = doc.xpathNewContext()

def fix(item):
	return item.content.replace("$PROJ_DIR$\\..\\", "").replace("$PROJ_DIR$/../", "").replace("\\","/")


if ((len(argv)==1) or (argv[1]=="src")):
	res = ctxt.xpathEval("//file[not(excluded[configuration='mksRobinNano'])]/name/text()")
	for e in res:
		e = fix(e)
		if os.path.isfile(e):
			print e
elif (argv[1]=="define"):
	res = ctxt.xpathEval("//settings/data/option[name='CCDefines']/state/text()")
	for e in res:
		print e
elif (argv[1]=="include"):
	res = ctxt.xpathEval("//settings/data/option[name='CCIncludePath2']/state/text()")
	for e in res:
		print fix(e)
elif (argv[1]=="lib"):
	res = ctxt.xpathEval("//settings/data/option[name='IlinkAdditionalLibs']/state/text()")
	for e in res:
		print fix(e)

